system/lib/wasmfs/file.h

```diff
@@ -150,22 +152,43 @@ class Directory : public File {
   };
 
 private:
+  enum class DCacheKind { Normal, Mount };
+  struct DCacheEntry {
+    DCacheKind kind;
```

Comment:
Trying to figure out the meaning of `kind` from the usage, it looks like a mount is another backend, is that right? It would be good to have a comment explaining that and what it means.